api/vector_tiles.py

```diff
@@ -58,7 +58,7 @@ def index(self, zoom, x, y):
         mapdb = cherrypy.request.app.config['DB']['map']
         d = mapdb.tables.style.data
 
-        q = sa.select([d.c.rels, d.c.allshields.label('shields'),
+        q = sa.select([d.c.rels, d.c.allrels, d.c.allshields.label('shields'),
```

Comment:
The join with the segments table is unnecessary when computing on the fly, I'd just do

  ... (select array(select distinct cycling.hierarchy.child FROM cycling.hierarchy WHERE cycling.hierarchy.parent=ANY(cycling.defstyle.rels))) as allrels, ...

(Note that you have to go down the hierarchy, as it's subrelations that are missing. Also `distinct` is necessary to eliminate duplicates.)

Good point about `allshields` though, maybe I'm just too paranoid about table space.